### DISCUSSION TITLE
Automatically cache FindByHashIdMixin Query

### DISCUSSION THREAD

**Discussion by bretterer** [deligoez/laravel-model-hashid#26] [Ideas]
It would be great if this package would cache the query for the FindByHashIdMixin package.  I am building an app using this package and currently have 12 calls to the database.

<img width="1613" alt="Screenshot 2024-01-12 at 9 40 14 PM" src="https://github.com/deligoez/laravel-model-hashid/assets/1906920/1dc318ef-7572-482f-a360-060bc70ae1de">

**Comment by bretterer** [comment]
Something like this should do it (and works for my testing

```
public function findByHashId(): Closure
    {
        /*
         * Find a model by its Hash Id.
         *
         * @param  mixed  $id
         * @param  array  $columns
         * @return \Illuminate\Database\Eloquent\Model|\Illuminate\Database\Eloquent\Collection|static[]|static|null
         */
        return function ($id, $columns = ['*']) {
            /* @var \Illuminate\Database\Eloquent\Builder $this */
            return Cache::remember($id, 10, function () use ($id, $columns) {
                return $this->find($this->getModel()->keyFromHashId($id), $columns);
            });
        };
    }
```

**Comment by deligoez** [comment]
Thank you for the suggestion and for sharing your caching solution!

We've decided to keep query caching out of the package scope — caching strategies vary significantly between applications (cache driver, TTL, invalidation, etc.) and are best handled at the application level, exactly as you demonstrated.

Your snippet is a great reference for anyone looking to add caching. Closing this as won't-fix, but the solution you shared works well.
